### ISSUE TITLE
Newer RenderDoc won't work with World of Warcraft

### ISSUE THREAD

**Issue by Vuhdo** [baldurk/renderdoc#484]
Latest RenderDoc fails to hook into World of Warcraft: Legion (running on Windows 10 if it matters).

I remember it worked fine in the past so I tried to find a working version. It turns out v0.28 works, but 0.29 doesn't. Looks like 0.29 can't recognize game's API:

`RENDERDOC:  [20:12:55]    remote_access.cpp( 390) - Log     - Got remote handshake: Wow-64 () [9968]`

... and then nothing more, while in 0.28 follows up with:

`RENDERDOC:  [20:14:08]    remote_access.cpp( 601) - Log     - Used API: D3D11`

**git bisect** helped to find out the exact commit when it stopped working:

```
"Add a LibraryHooks callback when capture options are changed"
https://github.com/baldurk/renderdoc/commit/20625eeebafdd4300f3bfeaf0548909f01
```

Specifically changes in hook.cpp:

https://github.com/baldurk/renderdoc/commit/20625eeebafdd4300f3bfeaf0548909f010148c7#diff-30fdc23c2ac57cc7548228548a1de6da

Restoring this check:

	if(!enable)
		return;

would make RenderDoc work again with World of Warcraft.
I don't know if this is a RenderDoc's error or WoW specific bug, but it would be great if you could investigate the issue. 

And btw, thanks a lot for your great work!

**Comment by baldurk** [comment]
This kind of looks like 'working as intended' actually. If `LibraryHooks::EnableHooks` is being called with `enable = false` (the only time the check would make a difference) then the program is requesting not to be captured via `D3DPERF_SetOptions`. RenderDoc will respect that and won't capture the program, so the fix in 20625ee is necessary. Without that the disabling of hooks was buggy and wouldn't propagate through properly.

You can double check by building yourself and see if `D3DPERF_SetOptions` is hit, but that seems like what's happening.

Obviously this is a bit artificial, as it's easy for you to compile a local copy with any such well-intentioned checks disabled, but I toe a bit of a careful line. While it's easy to work around, this isn't a bug and I'll always respect any program that doesn't want to be captured.

**Comment by Vuhdo** [comment]
Yeah, looks like that's the case.
Well, I guess I can live with that, thanks! 👍
